doc/src/manual/types.md

```diff
@@ -78,7 +70,10 @@ julia> (1+2)::Int
 ```
 
 This allows a type assertion to be attached to any expression in-place.
-
+When the type on the right is concrete,
+the value on the left must have that type as its implementation.
```

Comment:
```suggestion
```

this is true for dispatch purposes, but not true in general in cases like `x::Int = 'a'`